### PR TITLE
Api gatewat/feature/add get and update user info

### PR DESCRIPTION
**Pull request by ay0o5h**

### CODE REVIEW

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/UserDetailsDto.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.thechance.api_gateway.data.model
+
+import okhttp3.Address
+
+data class UserDetailsDto(
```
Comment (NadaFeteiha) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/UserDetailsDto.kt:5]: why you didn't use UserDto already in the API-gateway model?

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/data/service/IdentityService.kt [hunk 2]
```diff
@@ -101,14 +101,37 @@ class IdentityService(
...
     }
 
+    suspend fun updateUserProfile(
+        id: String,
+        fullName: String,
+        username: String,
+        email: String,
+        country: String,
+        languageCode: String
+    ): UserDetailsDto {
```
Comment (NadaFeteiha) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/data/service/IdentityService.kt:119]: The username and email are not allowed to be updated

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/UserRoute.kt [hunk 3]
```diff
@@ -39,6 +39,22 @@ fun Route.userRoutes() {
     }
 
     route("/user") {
+        authenticateWithRole(Role.END_USER) {
+          get("/{id}"){
+                val id = call.parameters["id"] ?: ""
```
Comment (NadaFeteiha) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/UserRoute.kt:44]: the ID should be extracted from the token.

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/UserRoute.kt [hunk 4]
```diff
@@ -39,6 +39,22 @@ fun Route.userRoutes() {
...
+                val language = extractLocalizationHeader()
+                val result = identityService.getUserById(id, language)
+                respondWithResult(HttpStatusCode.OK, result)
+          }
+            put("profile/{id}") {
```
Comment (NadaFeteiha) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/UserRoute.kt:49]: The ID should be extracted from the token.

**Review by NadaFeteiha — CHANGES_REQUESTED** [review]

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/UserDto.kt [hunk 5]
```diff
@@ -12,5 +12,5 @@ data class UserDto(
     @SerialName("country") val country: String,
     @SerialName("email") val email: String,
     @SerialName("permission") val permission: Int,
-    @SerialName("currency") val currency: String? = null,
+    @SerialName("currency") val currency: String? = null,z
```
Comment (mosayed01) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/UserDto.kt:15]:
```suggestion
    @SerialName("currency") val currency: String? = null,
```

**Review by mosayed01 — APPROVED** [review]

**Review by NadaFeteiha — CHANGES_REQUESTED** [review]
Screenshot 2023-09-23 at 8 50 43 AM

please test the branch and be sure that is run before making it ready for review. 
+ Add the endpoints in the description

**Review by devsadeq — APPROVED** [review]

**Review by aya-seif — APPROVED** [review]

**Review by NadaFeteiha — APPROVED** [review]

**Pull request merged**